### ISSUE TITLE
[GLPI 10] Errors on process RuleImportAsset

### ISSUE THREAD

**Issue by tsmr** [glpi-project/glpi#10708]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

Master

### Bug description

PHP Errors on OCS import 

```
[2022-02-18 16:45:30] glpisqllog.ERROR: DBmysql::query() in C:\wamp64\www\glpi\src\DBmysql.php line 360
  *** MySQL query error:
  SQL: SELECT `glpi_lines`.`id` FROM `glpi_lines` WHERE (`glpi_lines`.`serial` = 'VMware-56 4d ed e6 79 23 a4 78-6b 37 ae e1 57 de 16 84')
  Error: Champ 'glpi_lines.serial' inconnu dans where clause
  Backtrace :
  src\DBmysqlIterator.php:108                        
  src\DBmysql.php:1021                               DBmysqlIterator->execute()
  src\RuleImportAsset.php:537                        DBmysql->request()
  src\Rule.php:1641                                  RuleImportAsset->findWithGlobalCriteria()
  src\Rule.php:1499                                  Rule->checkCriterias()
  src\RuleCollection.php:1587                        Rule->process()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:563 RuleCollection->processAllRules()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:480 PluginOcsinventoryngOcsProcess::importComputer()
  plugins\ocsinventoryng\front\ocsng.import.php:93   PluginOcsinventoryngOcsProcess::processComputer()
  {"user":"2@P731"} 
[2022-02-18 16:45:30] glpisqllog.ERROR: DBmysql::query() in C:\wamp64\www\glpi\src\DBmysql.php line 360
  *** MySQL query error:
  SQL: SELECT `glpi_softwareversions`.`id` FROM `glpi_softwareversions` WHERE (`glpi_softwareversions`.`serial` = 'VMware-56 4d ed e6 79 23 a4 78-6b 37 ae e1 57 de 16 84')
  Error: Champ 'glpi_softwareversions.serial' inconnu dans where clause
  Backtrace :
  src\DBmysqlIterator.php:108                        
  src\DBmysql.php:1021                               DBmysqlIterator->execute()
  src\RuleImportAsset.php:537                        DBmysql->request()
  src\Rule.php:1641                                  RuleImportAsset->findWithGlobalCriteria()
  src\Rule.php:1499                                  Rule->checkCriterias()
  src\RuleCollection.php:1587                        Rule->process()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:563 RuleCollection->processAllRules()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:480 PluginOcsinventoryngOcsProcess::importComputer()
  plugins\ocsinventoryng\front\ocsng.import.php:93   PluginOcsinventoryngOcsProcess::processComputer()
  {"user":"2@P731","mem_usage":"0.011\", 6.63Mio)"} 
[2022-02-18 16:45:30] glpisqllog.ERROR: DBmysql::query() in C:\wamp64\www\glpi\src\DBmysql.php line 360
  *** MySQL query error:
  SQL: SELECT `glpi_clusters`.`id` FROM `glpi_clusters` WHERE (`glpi_clusters`.`serial` = 'VMware-56 4d ed e6 79 23 a4 78-6b 37 ae e1 57 de 16 84')
  Error: Champ 'glpi_clusters.serial' inconnu dans where clause
  Backtrace :
  src\DBmysqlIterator.php:108                        
  src\DBmysql.php:1021                               DBmysqlIterator->execute()
  src\RuleImportAsset.php:537                        DBmysql->request()
  src\Rule.php:1641                                  RuleImportAsset->findWithGlobalCriteria()
  src\Rule.php:1499                                  Rule->checkCriterias()
  src\RuleCollection.php:1587                        Rule->process()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:563 RuleCollection->processAllRules()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:480 PluginOcsinventoryngOcsProcess::importComputer()
  plugins\ocsinventoryng\front\ocsng.import.php:93   PluginOcsinventoryngOcsProcess::processComputer()
  {"user":"2@P731","mem_usage":"0.002\", 6.61Mio)"} 
[2022-02-18 16:45:30] glpisqllog.ERROR: DBmysql::query() in C:\wamp64\www\glpi\src\DBmysql.php line 360
  *** MySQL query error:
  SQL: SELECT `glpi_contracts`.`id` FROM `glpi_contracts` WHERE (`glpi_contracts`.`serial` = 'VMware-56 4d ed e6 79 23 a4 78-6b 37 ae e1 57 de 16 84')
  Error: Champ 'glpi_contracts.serial' inconnu dans where clause
  Backtrace :
  src\DBmysqlIterator.php:108                        
  src\DBmysql.php:1021                               DBmysqlIterator->execute()
  src\RuleImportAsset.php:537                        DBmysql->request()
  src\Rule.php:1641                                  RuleImportAsset->findWithGlobalCriteria()
  src\Rule.php:1499                                  Rule->checkCriterias()
  src\RuleCollection.php:1587                        Rule->process()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:563 RuleCollection->processAllRules()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:480 PluginOcsinventoryngOcsProcess::importComputer()
  plugins\ocsinventoryng\front\ocsng.import.php:93   PluginOcsinventoryngOcsProcess::processComputer()
  {"user":"2@P731","mem_usage":"0.002\", 6.61Mio)"} 
[2022-02-18 16:45:30] glpisqllog.ERROR: DBmysql::query() in C:\wamp64\www\glpi\src\DBmysql.php line 360
  *** MySQL query error:
  SQL: SELECT `glpi_databaseinstances`.`id` FROM `glpi_databaseinstances` WHERE (`glpi_databaseinstances`.`serial` = 'VMware-56 4d ed e6 79 23 a4 78-6b 37 ae e1 57 de 16 84')
  Error: Champ 'glpi_databaseinstances.serial' inconnu dans where clause
  Backtrace :
  src\DBmysqlIterator.php:108                        
  src\DBmysql.php:1021                               DBmysqlIterator->execute()
  src\RuleImportAsset.php:537                        DBmysql->request()
  src\Rule.php:1641                                  RuleImportAsset->findWithGlobalCriteria()
  src\Rule.php:1499                                  Rule->checkCriterias()
  src\RuleCollection.php:1587                        Rule->process()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:563 RuleCollection->processAllRules()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:480 PluginOcsinventoryngOcsProcess::importComputer()
  plugins\ocsinventoryng\front\ocsng.import.php:93   PluginOcsinventoryngOcsProcess::processComputer()
  {"user":"2@P731","mem_usage":"0.011\", 6.63Mio)"} 
[2022-02-18 16:45:30] glpisqllog.ERROR: DBmysql::query() in C:\wamp64\www\glpi\src\DBmysql.php line 360
  *** MySQL query error:
  SQL: SELECT `glpi_cables`.`id` FROM `glpi_cables` WHERE (`glpi_cables`.`serial` = 'VMware-56 4d ed e6 79 23 a4 78-6b 37 ae e1 57 de 16 84')
  Error: Champ 'glpi_cables.serial' inconnu dans where clause
  Backtrace :
  src\DBmysqlIterator.php:108                        
  src\DBmysql.php:1021                               DBmysqlIterator->execute()
  src\RuleImportAsset.php:537                        DBmysql->request()
  src\Rule.php:1641                                  RuleImportAsset->findWithGlobalCriteria()
  src\Rule.php:1499                                  Rule->checkCriterias()
  src\RuleCollection.php:1587                        Rule->process()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:563 RuleCollection->processAllRules()
  ...ins\ocsinventoryng\inc\ocsprocess.class.php:480 PluginOcsinventoryngOcsProcess::importComputer()
  plugins\ocsinventoryng\front\ocsng.import.php:93   PluginOcsinventoryngOcsProcess::processComputer()
  {"user":"2@P731","mem_usage":"0.003\", 6.61Mio)"} 
```

**Comment by cedric-anne** [comment]
It seems weird to me to base asset import rules on `$CFG_GLPI["state_types"]`, but I am not sure about which config entry should be used here.

@trasher ?

```diff
diff --git a/src/RuleImportAsset.php b/src/RuleImportAsset.php
index a6ffc4845f..e6bb1d47aa 100644
--- a/src/RuleImportAsset.php
+++ b/src/RuleImportAsset.php
@@ -471,7 +471,7 @@ class RuleImportAsset extends Rule
         ) {
             $itemtypeselected[] = $input['itemtype'];
         } else {
-            foreach ($CFG_GLPI["state_types"] as $itemtype) {
+            foreach ($CFG_GLPI["asset_types"] as $itemtype) {
                 if (
                     class_exists($itemtype)
                     && $itemtype != 'SoftwareLicense'
```

**Comment by tsmr** [comment]
I test this correction

**Comment by cedric-anne** [comment]
Fixed by #12043.
